### DISCUSSION TITLE
Run sample Application with NginxUnit + NJS module

### DISCUSSION THREAD

**Discussion by allwin-winfred** [nginx/unit#1259] [Support]
Hello team,

I configured a one backend application with nginunit plus njs module but how can I configure the njs with nginxunit,
I referred so many documents it won't work can you help me and guide the installation and deployment steps?

**Comment by lcrilly** [comment]
I'm not completely sure what you're trying to achieve.

Firstly, check that your Unit installation was built with njs. Run `unitd --version` and check that the configure options includes `--njs`.

Secondly, and just to clarify, njs is a tool for extending Unit's configuration - it is not intended for writing backend applications.
* The parts of Unit's configuration that can be extended are described in the [scripting docs](https://unit.nginx.org/scripting/).
* There is a simple example of creating a njs module for Unit in the [1.30 announcement](https://unit.nginx.org/news/2023/unit-1.30.0-released/#javascript-modules).

I hope that helps. If not, please elaborate on your goals.

**Reply by allwin-winfred** [reply]
Hi @lcrilly ,

I'm trying to prepare one sample example application with Nginx unit + njs this I'm trying,

And I check my unitd version,

`
unit version: 1.32.1
configured as ./configure --prefix=/usr --statedir=/var/lib/unit --control=unix:/var/run/control.unit.sock --runstatedir=/var/run --pid=/var/run/unit.pid --logdir=/var/log --log=/var/log/unit.log --tmpdir=/var/tmp --user=unit --group=unit --tests --openssl --njs --modulesdir=/usr/lib/unit/modules --libdir=/usr/lib/x86_64-linux-gnu --cc-opt='-g -O2 -fdebug-prefix-map=/data/builder/debuild/unit-1.32.1/pkg/deb/debuild/unit-1.32.1=. -specs=/usr/share/dpkg/no-pie-compile.specs -fstack-protector-strong -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fPIC' --ld-opt='-Wl,-Bsymbolic-functions -specs=/usr/share/dpkg/no-pie-link.specs -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie'`


Also, I need to configure the nginxunit + njs example

**Reply by lcrilly** [reply]
Please elaborate on your goals.

What do you want njs to do?

Can you share your current configuration?

**Reply by allwin-winfred** [reply]
Sure,

I have one node js application that needs to be authenticate using nginx unit + njs , I also need clarification regarding nginx unit scripting.

**Reply by lcrilly** [reply]
Do you want njs to perform the auth? What type of auth?

There's no point asking for clarification when you continue to provide minimal information and don't answer direct questions.

Please help me to help you.

**Reply by allwin-winfred** [reply]
yes, I want to implement Azure OpenID auth but I don't know how to perform also I don't have a configuration file.

**Reply by lcrilly** [reply]
That will be a large undertaking and it's quite likely you'll hit a blocker.

I recommend implementing OpenID Connect in your Node.JS application (there are several npm packages for this).

**Reply by allwin-winfred** [reply]
okay then I need an one sample application(whatever) with nginx unit + njs,
what are the files(code and configs) must I use on my server?
